model/span/event.go

```diff
@@ -447,5 +449,31 @@ func (e *Event) fields(tctx *transform.Context) common.MapStr {
 
 	st := e.Stacktrace.Transform(tctx)
 	utility.Set(fields, "stacktrace", st)
+
+	if h := e.servicemapHash(tctx); h != "" {
+		utility.Set(fields, "servicemap", common.MapStr{"fingerprint": h})
+	}
 	return fields
 }
+
+func (e *Event) servicemapHash(tctx *transform.Context) string {
+	if e.Destination == nil || e.Destination.Address == nil || *e.Destination.Address == "" {
+		return ""
+	}
+	h := xxhash.New()
+	fmt.Fprintf(h, "%s|%s", *e.Destination.Address, e.Type)
+	if e.Subtype != nil {
+		fmt.Fprintf(h, "|%s", *e.Subtype)
+	}
+	if e.Service != nil && e.Service.Name != nil {
```

Comment:
can this service name/environment be consolidated somewhere?  can it be done just once before transform?

Reply:
I changed the solution a bit. 

Regarding your comment that it should be done just once per transform - this is only run once per event transform. Am I missing something here?

Reply:
As we discussed IRL, i was specifically referring to the pick the first filled in service name or environment.  The change you already made is mostly what I meant.